DUnitX.Assert.pas

```diff
@@ -359,6 +361,23 @@ class procedure Assert.AreEqual(const expected, actual: boolean; const message:
     FailFmt(SUnexpectedErrorStr ,[BoolToStr(expected, true), BoolToStr(actual, true), message], ReturnAddress);
 end;
 
+class procedure Assert.AreEqual(const expected, actual: string; const compformat: TDUnitXComparableFormatClass; const ignoreCase: boolean; const message: string);
```

Comment:
Ok - I figured this would need some explaining.  This was the best way that I could come up with for a pluggable formatting identifier.  If I used an enum, code would have to be changed to hardcode in new formats (currently I put in XML and CSV).  Using a string identifier would be ok, except that then the override signature gets ambiguous.  I originally went with an integer, but again, there would have to be a const somewhere to ID it and basically would make the formats supported hardcoded in core files.

I decided instead to create a base class, then use the class as the identifier.  The class is not supposed to be instantiated - it is merely a fancy identifier (and strongly-typed one) that allows for flexibility down the road.  Currently I ONLY use the classes to ID the format of the data (i.e. XML) so BeyondCompare can use the XML compare instead of straight text.  However, later, the classes could actually implement some class procs or vars that customize things, like "compare with or without CRLF".

I know the naming is a bit heavy handed (I believe in being verbose on naming in large projects) but I ultimately think this was the most elegant way to handle identification of formats with the added benefit of giving a place to have format specific config later on.